Mathlib/CategoryTheory/FiberedCategory/Fibered.lean

```diff
@@ -0,0 +1,74 @@
+/-
+Copyright (c) 2024 Paul Lezeau. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Paul Lezeau, Calle Sönne
+-/
+
+import Mathlib.CategoryTheory.FiberedCategory.Cartesian
+
+/-!
+
+# Fibered categories
+
+This file defines what it means for a functor `p : 𝒳 ⥤ 𝒮` to be (pre)fibered.
+
+## Main definitions
+
+- `IsPreFibered p` expresses `𝒳` is fibered over `𝒮` via a functor `p : 𝒳 ⥤ 𝒮`, as in SGA VI.6.1.
+This means that any morphism in the base `𝒮` can be lifted to a cartesian morphism in `𝒳`.
+
+## Implementation
+
+The constructor of `IsPreFibered` is called `has_pullbacks'`. The reason for the prime is that when
+wanting to apply this condition, it is recommended to instead use the lemma `has_pullbacks`
+(without the prime), which is more applicable with respect to non-definitional equalities.
+
+## References
+* [A. Grothendieck, M. Raynaud, *SGA 1*](https://arxiv.org/abs/math/0206203)
+
+-/
+
+universe v₁ v₂ u₁ u₂
+
+open CategoryTheory Functor Category IsHomLift
+
+namespace CategoryTheory
+
+variable {𝒮 : Type u₁} {𝒳 : Type u₂} [Category.{v₁} 𝒮] [Category.{v₂} 𝒳]
+
+/-- Definition of a prefibered category.
+
+See SGA 1 VI.6.1. -/
+class Functor.IsPreFibered (p : 𝒳 ⥤ 𝒮) : Prop where
+  has_pullbacks' {a : 𝒳} {R : 𝒮} (f : R ⟶ p.obj a) : ∃ (b : 𝒳) (φ : b ⟶ a), IsCartesian p f φ
```

Comment:
Perhaps should be called `exists_isCartesian'` (and similarly for the lemma below)?

Reply:
Good suggestion! Fixed :)